src/main/kotlin/app/ui/AuthState.kt

```diff
@@ -12,19 +12,21 @@ import app.utils.PasswordHelper
  */
 class AuthState constructor(val context: Context) : ConsoleState {
     override fun doAction() {
-        var isCredentialsUnknown = Configurator.config.username.isEmpty()
-                || Configurator.config.password.isEmpty()
+        var username = Configurator.config.username
+        var password = Configurator.config.password
+        var isCredentialsUnknown = username.isEmpty() || password.isEmpty()
```

Comment:
Discovering of configurator is a violation of dependency assignation. See GRASP.